hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/GaugeImpl.java

```diff
@@ -0,0 +1,150 @@
+    volatile S source;
+
+    private final ILogger logger;
+    private final String name;
+
+    public GaugeImpl(String name, ILogger logger) {
+        this.name = name;
+        this.logger = logger;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public void render(StringBuilder sb) {
+        checkNotNull(sb, "sb can't be null");
+
+        sb.append(name).append('=');
+
+        Object input = this.input;
+        S source = this.source;
+
+        if (input == null || source == null) {
+            sb.append("NA");
+            return;
+        }
+
+        try {
+            if (input instanceof LongProbe) {
+                LongProbe function = (LongProbe) input;
+                sb.append(function.get(source));
+            } else if (input instanceof DoubleProbe) {
+                DoubleProbe function = (DoubleProbe) input;
+                sb.append(function.get(source));
+            } else if (input instanceof AccessibleObjectGauge) {
+                AccessibleObjectGauge accessibleObjectInput = (AccessibleObjectGauge) input;
+                if (accessibleObjectInput.isDouble()) {
+                    sb.append(accessibleObjectInput.getDouble(source));
+                } else {
+                    sb.append(accessibleObjectInput.getLong(source));
+                }
+            }
+        } catch (Exception e) {
+            logger.warning("Failed to update metric:" + name, e);
+            sb.append("NA");
+        }
+    }
+
+    @Override
+    public long readLong() {
+        Object input = this.input;
+        S source = this.source;
+        long result = 0;
+
+        if (input == null || source == null) {
+            return result;
+        }
+
+        try {
+            if (input instanceof AccessibleObjectGauge) {
+                AccessibleObjectGauge accessibleObjectInput = (AccessibleObjectGauge) input;
+                if (accessibleObjectInput.isDouble()) {
+                    double doubleResult = accessibleObjectInput.getDouble(source);
+                    result = Math.round(doubleResult);
+                } else {
+                    result = accessibleObjectInput.getLong(source);
+                }
+            } else if (input instanceof LongProbe) {
+                LongProbe<S> longInput = (LongProbe<S>) input;
+                result = longInput.get(source);
+            } else if (input instanceof DoubleProbe) {
+                DoubleProbe<S> doubleInput = (DoubleProbe<S>) input;
+                double doubleResult = doubleInput.get(source);
+                result = Math.round(doubleResult);
+            }
+            return result;
+        } catch (Exception e) {
+            logger.warning("Failed to update metric:" + name, e);
+            return result;
+        }
+    }
+
+    @Override
+    public double readDouble() {
+        Object input = this.input;
+        S source = this.source;
+        double result = 0;
```

Comment:
Assuming we leave the current LongProbe and DoubleProbe interface as they are, then the following code could be simplified to:

before
```
 @Override
    public long readLong() {
        Object input = this.input;
        S source = this.source;
        long result = 0;

        if (input == null || source == null) {
            return result;
        }

        try {
            if (input instanceof AccessibleObjectProbe) {
                AccessibleObjectProbe accessibleObjectInput = (AccessibleObjectProbe) input;
                if (accessibleObjectInput.isDouble()) {
                    double doubleResult = accessibleObjectInput.getDouble(source);
                    result = Math.round(doubleResult);
                } else {
                    result = accessibleObjectInput.getLong(source);
                }
            } else if (input instanceof LongProbe) {
                LongProbe<S> longInput = (LongProbe<S>) input;
                result = longInput.get(source);
            } else if (input instanceof DoubleProbe) {
                DoubleProbe<S> doubleInput = (DoubleProbe<S>) input;
                double doubleResult = doubleInput.get(source);
                result = Math.round(doubleResult);
            }
            return result;
        } catch (Exception e) {
            logger.warning("Failed to update metric:" + name, e);
            return result;
        }
    }
```

after
```
    @Override
    public long readLong() {
        Object input = this.input;
        S source = this.source;
        long result = 0;

        if (input == null || source == null) {
            return result;
        }

        try {
           if (input instanceof LongProbe) {
                LongProbe<S> longInput = (LongProbe<S>) input;
                result = longInput.get(source);
            } else if (input instanceof DoubleProbe) {
                DoubleProbe<S> doubleInput = (DoubleProbe<S>) input;
                double doubleResult = doubleInput.get(source);
                result = Math.round(doubleResult);
            }
            return result;
        } catch (Exception e) {
            logger.warning("Failed to update metric:" + name, e);
            return result;
        }
    }
```